### PR TITLE
[fix](systable) fix unstable case for partitions table

### PR DESCRIPTION
**Pull request by morningman**
Followup #40153

### CODE REVIEW

**Comment by morningman** [comment]
run buildall

File: be/src/exec/schema_scanner/schema_scanner_helper.cpp [hunk 1]
```diff
@@ -50,8 +50,8 @@ void SchemaScannerHelper::insert_datetime_value(int col_index, const std::vector
     nullable_column->get_null_map_data().emplace_back(0);
 }
 
-void SchemaScannerHelper::insert_int_value(int col_index, int64_t int_val,
-                                           vectorized::Block* block) {
+void SchemaScannerHelper::insert_int64_value(int col_index, int64_t int_val,
+                                             vectorized::Block* block) {
     vectorized::MutableColumnPtr mutable_col_ptr;
     mutable_col_ptr = std::move(*block->get_by_position(col_index).column).assume_mutable();
     auto* nullable_column = reinterpret_cast<vectorized::ColumnNullable*>(mutable_col_ptr.get());
```
Comment (yiguolei) [inline be/src/exec/schema_scanner/schema_scanner_helper.cpp:57]: better to use assert cast instead of reinterpret_cast

Comment (wangbo) [reply be/src/exec/schema_scanner/schema_scanner_helper.cpp:57]: assert_cast could log fatal may cause core dump，how about check type and return status::error ehre

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

File: regression-test/suites/query_p0/system/test_partitions_schema.groovy [hunk 2]
```diff
@@ -184,12 +184,12 @@ suite("test_partitions_schema") {
...
     sql "GRANT SELECT_PRIV ON ${dbName}.duplicate_table  TO ${user}"
     connect(user=user, password='123abc!@#', url=url) {
-        qt_select_check_4 """select  $listOfColum from information_schema.partitions where table_schema=\"${dbName}\" order by $listOfColum"""
+        order_qt_select_check_4 """select TABLE_CATALOG,TABLE_SCHEMA,TABLE_NAME,PARTITION_NAME from information_schema.partitions where table_schema=\"${dbName}\""""
     }
     
     sql "REVOKE SELECT_PRIV ON ${dbName}.duplicate_table  FROM ${user}"
     connect(user=user, password='123abc!@#', url=url) {
-        qt_select_check_5 """select  $listOfColum from information_schema.partitions where table_schema=\"${dbName}\" order by $listOfColum"""
+        order_qt_select_check_5 """select TABLE_CATALOG,TABLE_SCHEMA,TABLE_NAME,PARTITION_NAME from information_schema.partitions where table_schema=\"${dbName}\""""
```
Comment (wangbo) [inline regression-test/suites/query_p0/system/test_partitions_schema.groovy:192]: test sql missing order by

**Review by wangbo — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]

**Pull request merged**